src/Widgets/LinkListBox.vala

```diff
@@ -0,0 +1,286 @@
+        if (contact_url != null) {
+            contact_listbox.append (new LinkRow (
+                contact_url,
+                _("Contact The Developer"),
+                "mail-send-symbolic",
+                "yellow"
+            ));
+        }
+
+        var contribute_listbox = new Gtk.ListBox () {
+            accessible_role = GENERIC,
+            hexpand = true,
+            show_separators = true,
+            selection_mode = NONE,
+            valign = START
+        };
+        contribute_listbox.add_css_class ("boxed-list");
+        contribute_listbox.add_css_class (Granite.STYLE_CLASS_RICH_LIST);
+
+        var project_license = component.project_license;
+        if (project_license != null) {
+            string? license_label = null;
+            string? license_url = null;
+            parse_license (project_license, out license_label, out license_url);
+
+            contribute_listbox.append (new LinkRow (
+                license_url,
+                _(license_label),
+                "text-x-copying-symbolic",
+                "slate"
+            ));
+        }
+
+        var sponsor_url = component.get_url (DONATION);
+#if PAYMENTS
+        var payments_key = component.get_custom_value ("x-appcenter-stripe");
+        if (payments_key != null) {
+            sponsor_url = payments_key;
+        }
+#endif
+        if (sponsor_url != null) {
+            contribute_listbox.append (new LinkRow (
+                sponsor_url,
+                _("Sponsor"),
+                "face-heart-symbolic",
+                "pink"
+            ));
+        }
+
+        var translate_url = component.get_url (TRANSLATE);
+        if (translate_url != null) {
+            contribute_listbox.append (new LinkRow (
+                translate_url,
+                _("Contribute Translations"),
+                "preferences-desktop-locale-symbolic",
+                "blue"
+            ));
+        }
+
+        var vcs_url = component.get_url (VCS_BROWSER);
+        if (vcs_url != null) {
+            contribute_listbox.append (new LinkRow (
+                vcs_url,
+                _("Get The Source Code"),
+                "link-vcs-symbolic",
+                "purple"
+            ));
+        }
+
+        var flowbox = new Gtk.FlowBox () {
+            column_spacing = 24,
+            max_children_per_line = 2,
+            row_spacing = 24,
+            selection_mode = NONE
+        };
+        flowbox.set_parent (this);
+
+        if (contribute_listbox.get_first_child () != null) {
+            flowbox.append (contribute_listbox);
+        }
+
+        if (contact_listbox.get_first_child () != null) {
+            flowbox.append (contact_listbox);
+        }
+
+        // Don't let invisible container get focus
+        var flowbox_child = flowbox.get_first_child ();
+        while (flowbox_child != null) {
+            flowbox_child.focusable = false;
+            flowbox_child = flowbox_child.get_next_sibling ();
+        }
```

Comment:
I'm not sure what this actually does. The comment says that this prevents invisible flowboxes from being focused but the code doesn't check for the visibility of flowboxes.

Reply:
Updated! Is that better?

Reply:
Thank you, now I understand what we're doing here.